src/Sarif.Viewer.VisualStudio.Core/SarifViewerPackage.cs

```diff
@@ -203,12 +215,60 @@ private void SolutionEvents_OnBeforeCloseSolution(object sender, EventArgs e)
         {
             // stop watcher when the solution is closed.
             this.sarifFolderMonitor?.StopWatch();
+
+            var fileSystem = new FileSystem();
+
+            try
+            {
+                fileSystem.FileDelete(Path.Combine(GetDotSarifDirectoryPath(), "scan-results.sarif"));
+            }
+            catch (Exception) { }
         }
 
         private void SolutionEvents_OnAfterBackgroundSolutionLoadComplete(object sender, EventArgs e)
         {
             // start to watch when the solution is loaded.
             this.sarifFolderMonitor?.StartWatch();
+
+            this.JoinableTaskFactory.Run(async () => await RequestAnalysisResultsAsync());
+        }
+
+        private async Task RequestAnalysisResultsAsync()
```

Comment:
>RequestAnalysisResultsAsync [](http://example.com/codeflow?start=27&length=27)

nit: would create a wrapper class to deal with ResultSourceService call/event handling, make the vs package code short, what do you think